exist-core/src/main/java/org/exist/xquery/functions/map/MapType.java

```diff
@@ -216,6 +216,24 @@ public boolean contains(AtomicValue key) {
         return map.contains(key);
     }
 
+    @Override
+    public boolean containsReference(final Item item) {
+        for (final Iterator<Sequence> it = map.values().iterator(); it.hasNext();) {
+            final Sequence value = it.next();
+            return value == item;
```

Comment:
I guess this will not work correctly f there are more than one sequence content and should be written somewhat like this:
```
if (value == item)) {
    return true;
}
```

Reply:
Okay fixed.